### PR TITLE
feat: add landscape capability

### LINKED ISSUE #7: Usable as landscape also
**Describe the feature you'd like**

Now the application runs only for potrait (declared as potrait in manifests). To make it also work in landscape mode.

**Would you like to work on the issue?**

No

### PR DESCRIPTION
**Pull request by adarshsiva**
Fixed: #7 

**Changes:**

Added landscape compatibility. 

**Screenshots:**
Landscape:
Webp net-resizeimage (1)
Webp net-resizeimage (3)

Webp net-resizeimage (4)
Portrait:
Webp net-resizeimage (7)

Webp net-resizeimage (6)

Webp net-resizeimage (5)

### CODE REVIEW

**Comment by immadisairaj** [comment]
@adarshsiva please follow semantic PR guidelines.
https://github.com/commitizen/conventional-commit-types/blob/master/index.json

**Comment by immadisairaj** [comment]
@adarshsiva can you also add the screenshot of the activity you've changed and also in landscape mode

**Comment by immadisairaj** [comment]
@adarshsiva the title should be
`feat: add landscape capability`

**Comment by immadisairaj** [comment]
@adarshsiva don't remove the pr template in the comment.
You add screenshot along with the template given.

**Comment by immadisairaj** [comment]
It has to be in that format.
https://github.com/commitizen/conventional-commit-types/blob/master/index.json
this link is doc for what kind of pr. If it is bug you write `fix:` and if it is feature you write `feat:` and so on

**Comment by adarshsiva** [comment]
@immadisairaj  please review

**Comment by immadisairaj** [comment]
@adarshsiva, when I rotate my phone to landscape or portrait mode in info activity the whole, is loading again. Instead it should load and load again and use the already stored value.

**Comment by immadisairaj** [comment]
also can make the grid to 3 when in landscape in submissions

> when I rotate my phone to landscape or portrait mode in info activity the whole, is loading again. Instead it should load and load again and use the already stored value.

same thing is applicable in submission activity also

Edit:
I think the UI is not soo good and optimised. if you can improve it then you can do that also.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -13,7 +13,8 @@
         android:theme="@style/AppTheme">
         <activity
             android:name="com.example.immadisairaj.codeforces.MainActivity"
-            android:screenOrientation="portrait">
+            android:configChanges="orientation"
+            >
```
Comment (immadisairaj) [inline app/src/main/AndroidManifest.xml]: this can go to the above line

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -23,11 +24,13 @@
         <activity
             android:name="com.example.immadisairaj.codeforces.InfoActivity"
             android:parentActivityName="com.example.immadisairaj.codeforces.MainActivity"
-            android:screenOrientation="portrait" />
+            android:configChanges="orientation"
+             />
```
Comment (immadisairaj) [inline app/src/main/AndroidManifest.xml]: same as above

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -23,11 +24,13 @@
...
         <activity
             android:name="com.example.immadisairaj.codeforces.SubmissionActivity"
             android:parentActivityName="com.example.immadisairaj.codeforces.InfoActivity"
-            android:screenOrientation="portrait" />
+            android:configChanges="orientation"
+             />
```
Comment (immadisairaj) [inline app/src/main/AndroidManifest.xml]: same as above

**Review by immadisairaj — CHANGES_REQUESTED** [review]

**Comment by adarshsiva** [comment]
Requested Changes have been made.

**Comment by immadisairaj** [comment]
@adarshsiva can you update the screenshots and also add gif if possible

Edit:
both in portrait and landscape

**Comment by immadisairaj** [comment]
@adarshsiva can you resolve the conflicts

**Comment by adarshsiva** [comment]
I'll  update and revert back.

**Comment by immadisairaj** [comment]
> I'll update and revert back.

don't revert. just resolve

**Comment by adarshsiva** [comment]
Screenshots have been added.

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -13,7 +13,7 @@
         android:theme="@style/AppTheme">
         <activity
             android:name="com.example.immadisairaj.codeforces.MainActivity"
-            android:screenOrientation="portrait">
+            android:configChanges="orientation|screenSize">
```
Comment (immadisairaj) [inline app/src/main/AndroidManifest.xml:16]: what is the use of this?

Comment (adarshsiva) [reply app/src/main/AndroidManifest.xml:16]: That is used to manually handle configuration changes in an activity. If we want to manually handle configuration changes in the app we must declare both the "orientation" and "screenSize" values in the android:configChanges attributes.

Comment (immadisairaj) [reply app/src/main/AndroidManifest.xml:16]: `screenSize` is needed?

Comment (adarshsiva) [reply app/src/main/AndroidManifest.xml:16]: Yes

Comment (adarshsiva) [reply app/src/main/AndroidManifest.xml]: [Reference](https://developer.android.com/guide/topics/resources/runtime-changes)

**Review by immadisairaj — APPROVED** [review]

**Pull request merged**